assets/account/src/lib.rs

```diff
@@ -33,7 +33,7 @@ blueprint! {
         pub fn deposit(&mut self, bucket: Bucket) {
             let address = bucket.resource_address();
             match self.vaults.get(&address) {
-                Some(v) => {
+                Some(mut v) => {
```

Comment:
This might slightly go beyond the scope of the original task, but I think it would be nice if we could take the chance to refactor the implementation of lazy map to be consistent to `HashMap` from std.

Specifically, `get()` should return an immutable reference to the value and `get_mut` should return a mutable reference.

Because of the current implementation, I've seen some bad coding pattern (get-update-insert) from myself and the community.